applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -121,19 +115,21 @@ void BaseSolidElement::InitializeSolutionStep( const ProcessInfo& rCurrentProces
         const GeometryType::IntegrationPointsArrayType& integration_points = this->IntegrationPoints();
 
         for ( IndexType point_number = 0; point_number < mConstitutiveLawVector.size(); ++point_number ) {
-            if (mConstitutiveLawVector[point_number]->RequiresInitializeMaterialResponse()) {
-                // Compute element kinematics B, F, DN_DX ...
-                CalculateKinematicVariables(this_kinematic_variables, point_number, mThisIntegrationMethod);
+            // Compute element kinematics B, F, DN_DX ...
+            CalculateKinematicVariables(this_kinematic_variables, point_number, mThisIntegrationMethod);
 
-                // Compute constitutive law variables
-                SetConstitutiveVariables(this_kinematic_variables, this_constitutive_variables, Values, point_number, integration_points);
+            // Compute constitutive law variables
+            SetConstitutiveVariables(this_kinematic_variables, this_constitutive_variables, Values, point_number, integration_points);
 
-                // Call the constitutive law to update material variables
-                mConstitutiveLawVector[point_number]->InitializeMaterialResponse(Values, GetStressMeasure());
+            // Rotate to local axes strain/F
+            if (is_rotated)
```

Comment:
what are u trying to do here?

Reply:
the main issue is that the check `IsElementRotated()` was done every time the `CalculateConstitutiveVariables` (so at each IP) and that was more expensive than putting the check outside the IP loop. I have an idea to encapsulate this better